extract/extract/extractors/amazon.py

```diff
@@ -0,0 +1,321 @@
+    Helper function that extracts the product's image urls.
+
+    Args:
+        soup (BeautifulSoup): Parsed HTML
+
+    Returns:
+        Optional[list[str]]: `list` object containing `str` objects representing the image urls.
+            If nothing was found `None` or empty list is returned.
+    """
+    targets = [
+        soup.find("div", {"id": "altImages"}).find_all("img"),
+    ]
+
+    def parse_image_urls(images: list[BeautifulSoup]) -> list[str]:
+        image_urls = [
+            image["src"]
+            for image in images
+            if not image["src"].endswith(".gif")
+            and "play-button-overlay" not in image["src"]
+            and "play-icon-overlay" not in image["src"]
+            and "360_icon" not in image["src"]
+        ]
+        return image_urls
+
+    return _handle_parse(targets, parse_image_urls)
+
+
+def _get_sizes(soup: BeautifulSoup) -> Optional[str]:
+    """
+    Helper function that extracts the product's sizes.
+
+    Args:
+        soup (BeautifulSoup): Parsed HTML
+
+    Returns:
+        Optional[str]: `str` object containing all sizes. If nothing was found `None`.
+    """
+    targets = [
+        soup.find_all("span", {"class", "a-size-base swatch-title-text-display swatch-title-text"})[
+            1:
+        ],
+        soup.find_all("option", id=re.compile("size_name"))[1:],
+    ]
+
+    def parse_sizes(sizes: list[BeautifulSoup]) -> str:
+        sizes = [size.text.strip() for size in sizes if size.text.strip()]
+        return ", ".join(sizes)
+
+    return _handle_parse(targets, parse_sizes)
+
+
+def _get_price(parsed_page: ParsedPage) -> Optional[float]:
+    """
+    Helper function that extracts the product's price.
+
+    Args:
+        parsed_page (ParsedPage): Intermediate representation of `ScrapedPage` domain object
+
+    Returns:
+        Optional[float]: `float` object if a price is given, else `None`.
+    """
+    targets = [
+        parsed_page.scraped_page.meta_information["price"],
+    ]
+
+    def parse_price(price: str) -> float:
+        return float(price.replace(".", "").replace(",", "."))
+
+    return _handle_parse(targets, parse_price)
+
+
+def _get_brand(soup: BeautifulSoup) -> Optional[str]:
+    """
+    Helper function that extracts the product's brand.
+
+    Args:
+        soup (BeautifulSoup): Parsed HTML
+
+    Returns:
+        Optional[str]: `str` object if a brand was found, else `None`.
+    """
+    targets = [
+        soup.find("div", {"id": "bylineInfo_feature_div"}).a.text,
+    ]
+
+    def parse_brand(brand: str) -> Optional[str]:
+        if brand.startswith("Besuche den "):
+            return brand[len("Besuche den ") : -len("-Store")]  # noqa
+        if brand.startswith("Marke: "):
+            return brand[len("Marke: ") :]  # noqa
+        return None
+
+    return (
+        _handle_parse(targets, parse_brand)
+        or _find_from_details_section(soup, "Marke")
+        or _find_from_details_section(soup, "Hersteller")
```

Comment:
Any chance that we do not use German words here? This will definitely break for the other languages.
Are these special cases or common?

Reply:
The UK brand strings seem to be roughly the same as these ones. For France, I see that `"Visit the X Store"` turns into `"Visiter la boutique X"`. Not sure what is best. Maybe we change the parsing function entirely based on a language param to the getter?

Reply:
So this type of page is common? Could you send some example URLs?